src/ui/ProductManagement.ts

```diff
@@ -0,0 +1,143 @@
+import CustomElement from './CustomElement';
+import TEMPLATE from '../templates';
+import { $, addEvent, deleteSeparator, emit, markUnit } from '../utils';
+import VendingMachine from '../domain/VendingMachine';
+import Product from '../domain/Product';
+import storage from '../storage';
+import { ELEMENT_KEY } from '../constants';
+
+class ProductManagement extends CustomElement {
+  connectedCallback() {
+    super.connectedCallback();
+    VendingMachine.instance.observe(ELEMENT_KEY.PRODUCT, this);
+  }
+
+  render() {
+    this.innerHTML = this.template();
+
+    const products = storage.getProducts();
+
+    products.forEach((product) => this.insertItem(product));
+  }
+
+  template() {
+    return TEMPLATE.PRODUCT_MANAGEMENT;
+  }
+
+  setEvent() {
+    addEvent(this, 'submit', '.product-manage-form', (e) => this.handleAdd(e));
+    addEvent(this, 'click', '.product-item', (e) => this.handleUpdateAndDelete(e));
+    addEvent(this, 'submit', '.product-item__form', (e) => this.handleConfirm(e));
+  }
+
+  handleAdd(e) {
+    e.preventDefault();
+
+    const name = e.target.name.value;
+    const price = e.target.price.valueAsNumber;
+    const quantity = e.target.quantity.valueAsNumber;
+
+    e.target.reset();
+
+    emit('.product-manage-form', '@add', { name, price, quantity }, this);
+  }
+
+  handleUpdateAndDelete(e) {
+    if (e.target.classList.contains('product-item__edit-button')) {
+      this.showForm(e);
+    }
+
+    if (e.target.classList.contains('product-item__delete-button') && confirm('해당 상품을 삭제하시겠습니까?')) {
+      const productName = e.target.closest('.product-item').dataset.productName;
+
+      emit('#product-list-table', '@delete', { productName }, this);
+    }
+  }
+
+  showForm(e) {
+    const item = e.target.closest('.product-item');
+    const { productName, productId } = item.dataset;
+    const values = [...item.getElementsByTagName('td')].slice(0, 3).map((td) => td.textContent);
+
+    const name = values[0];
+    const price = deleteSeparator(values[1]);
+    const quantity = values[2];
```

Comment:
```suggestion
    const [name, priceWithSeparator, quantity] = [...item.getElementsByTagName('td')].slice(0, 3).map((td) => td.textContent);

    const price = deleteSeparator(priceWithSeperator);
```
3번 배열이 들어온다고 확신할 수 있다면 이렇게도 해볼 수 있겠네요

Reply:
price 변수만 함수 호출이 필요해서 어떻게 리팩토링 할 수 있을지 고민했었는데
map으로 한 번 돌린 다음에 price만 따로 deleteSeperator 하는 방법도 있었네요~!
원래 고민했던 방식은 map 돌리면서 if 문으로 price일 경우에만 deleteSeperator 하는 방식이었는데 뭔가 지저분한 것 같아 지우고 위와 같은 코드를 작성했었습니다~ ㅠㅠ

Reply:
반영 링크입니다! [dd24d8a](https://github.com/woowacourse/javascript-vendingmachine/pull/32/commits/dd24d8aa94b05b8d8660d98b487525665a9c504e)